src/components/DropdownMenu/DropdownMenuPopup.tsx

```diff
@@ -31,30 +40,85 @@ export const DropdownMenuPopup = <T,>({
     size,
     menuProps,
     children,
+    popupProps,
+    path = [],
 }: DropdownMenuPopupProps<T>) => {
-    const content = useMemo(() => {
-        return (
-            children || (
-                <Menu className={b('menu')} size={size} {...menuProps}>
-                    {toItemList(items, SEPARATOR).map((item, index) => {
-                        const {className, ...itemProps} = item;
+    const {toggle, data} = useContext(DropdownMenuContext);
 
-                        return (
-                            <DropdownMenuItemComponent
-                                key={index}
-                                className={b(
-                                    'menu-item',
-                                    {separator: item === SEPARATOR},
-                                    className,
-                                )}
-                                {...itemProps}
-                            />
-                        );
-                    })}
-                </Menu>
-            )
-        );
-    }, [children, size, menuProps, items]);
+    const {
+        activeMenuPath,
+        setActiveMenuPath,
+        anchorRef: navigationAnchorRef,
+    } = useContext(DropdownMenuNavigationContext);
+
+    const isSubmenu = path.length > 0;
+
+    const activateParent = useCallback(() => {
+        setActiveMenuPath(path.slice(0, path.length - 1));
+    }, [setActiveMenuPath, path]);
+
+    const handleMouseEnter = useCallback(
```

Comment:
Our hover components is used different delay behaviors on hover, it would be nice to have it here